sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpContentRequestStreamHandle.cs

```diff
@@ -0,0 +1,112 @@
+{
+#if NET8_0_OR_GREATER
+    /// <summary>
+    /// This class is set as the Content for the HttpRequestMessage when a request is going
+    /// to use a IHttpRequestStreamPublisher to stream data to the service.
+    /// 
+    /// This is typically done for bi-directional event streaming operations that 
+    /// need to stream events from the consumer code to the service.
+    /// </summary>
+    public class HttpContentRequestStreamHandle : HttpContent, IHttpRequestStreamHandle
+    {
+        private Logger _logger = Logger.GetLogger(typeof(HttpContentRequestStreamHandle));
+        private HttpRequestMessage _httpRequest;
+        private bool _disposed;
+        private TaskCompletionSource _tcs = new TaskCompletionSource();
+        private IHttpRequestStreamPublisher _publisher;
+
+        public HttpContentRequestStreamHandle(HttpRequestMessage httpRequest, IHttpRequestStreamPublisher publisher)
+        {
+            _httpRequest = httpRequest;
+            _httpRequest.Content = this;
+
+            _publisher = publisher;
+        }
+
+        protected override async Task SerializeToStreamAsync( Stream stream, TransportContext context)
+        {
+            if(_publisher != null)
+            {
+                Byte[] bytes;
+                while((bytes = await _publisher.NextBytesAsync().ConfigureAwait(false)) != null)
+                {
+                    await stream.WriteAsync(bytes, 0, bytes.Length).ConfigureAwait(false);
+                    await stream.FlushAsync().ConfigureAwait(false);
+                }
+            }
+
+            // Even if the user has ended input streaming via returning null from the publisher
+            // we need to block the return here till the full operation is complete. Otherwise
+            // HttpClient will end the streaming session before the response has been completed streaming.
+            // This can cause trigger an error with the service. The TaskCompletionSource, _tcs, 
+            // is completed as part of disposing this class.
+            await _tcs.Task.ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Since the length is unknown based on the user continually streaming data return back false and -1 for length.
+        /// </summary>
+        /// <param name="length"></param>
+        /// <returns></returns>
+        protected override bool TryComputeLength(out long length)
+        {
+            length = -1; // Unknown length
+            return false;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            _logger.DebugFormat("Disposing the request stream");
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                if (_tcs != null)
+                {
+                    _logger.DebugFormat("Completed writing to request stream");
+                    _tcs.SetResult();
+                    _tcs = null;
```

Comment:
This is nulling out _tcs but it is possible the await above isn't done after it is signaled.